sdk/keyvault/azkeys/crypto/crypto_client.go

```diff
@@ -173,52 +184,55 @@ func (c *Client) Encrypt(ctx context.Context, alg EncryptionAlgorithm, value []b
 		c.vaultURL,
 		c.keyID,
 		c.keyVersion,
-		options.toGeneratedKeyOperationsParameters(alg, value),
+		options.toGeneratedKeyOperationsParameters(alg, plaintext),
 		&generated.KeyVaultClientEncryptOptions{},
 	)
 	if err != nil {
 		return EncryptResponse{}, err
 	}
 
-	return encryptResponseFromGenerated(resp), nil
+	return encryptResponseFromGenerated(resp, alg), nil
 }
 
 // DecryptOptions contains the optional parameters for the Client.Decrypt function
 type DecryptOptions struct {
 	// Additional data to authenticate but not encrypt/decrypt when using authenticated crypto algorithms.
-	AAD []byte `json:"aad,omitempty"`
+	AuthData []byte `json:"aad,omitempty"`
 
 	// Initialization vector for symmetric algorithms.
 	IV []byte `json:"iv,omitempty"`
 
 	// The tag to authenticate when performing decryption with an authenticated algorithm.
-	Tag []byte `json:"tag,omitempty"`
+	AuthTag []byte `json:"tag,omitempty"`
```

Comment:
If AuthTag and AuthData are related you could place them together in the struct.. I don't know if we alpha-sort but that would have done it too.